core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -86,6 +99,9 @@
 public class CatalogHandlers {
   private static final Schema EMPTY_SCHEMA = new Schema();
   private static final String INITIAL_PAGE_TOKEN = "";
+  private static final InMemoryPlanningState IN_MEMORY_PLANNING_STATE =
+      InMemoryPlanningState.getInstance();
```

Comment:
this looks much nicer. I wonder whether we should actually push all of the server-side scan planning state + behavior into a wrapper class. That way also `CatalogHandlers` becomes cleaner and you have everything related to server-side scan planning in a single class. This isn't a blocker but I was just curious how that would look